### PR TITLE
Rewrite team invites - Closes #103

### LINKED ISSUE #103: Team Invites functionality isn't organized
TeamInvite objects are stored and used to track membership on a team, however it'd be much easier to just check the user exists in the teams founder fk field, captains mtm field, or the players mtm field. 

Seeing as this is a rather large fix porting a site from the previous work arounds could cause issues, would like this completed sooner rather then later to avoid compatability issues.

### PR DESCRIPTION
**Pull request by mikemaddem**
Rewrite team invites to be used as strictly invites and not as membership within a team

### CODE REVIEW

**Comment by mikemaddem** [comment]
merging into notifications so the work on notifications can be completed

File: staff/views/teams.py [hunk 1]
```diff
@@ -61,6 +64,7 @@ def delete_team(request, pk):
         return redirect('staff:teamindex')
 
 
+#TODO remove TeamInvite object
```
Comment (techlover1) [inline staff/views/teams.py:67]: Resolve this TODO

File: teams/views.py [hunk 2]
```diff
@@ -31,7 +33,8 @@ class MyInvitesListView(ListView):
     model = TeamInvite
 
     def get(self, request):
-        teaminvite_list = TeamInvite.objects.filter(user=self.request.user, active=True)
+        teaminvite_list = TeamInvite.objects.filter(Q(user=self.request.user, active=True))
+        # teaminvite_list = teaminvite_list.filter(Q(fou))
```
Comment (techlover1) [inline teams/views.py:37]: Get rid of this line

File: teams/views.py [hunk 3]
```diff
@@ -83,28 +88,33 @@ class MyTeamsListView(ListView):
...
-            return TeamInvite.objects.filter(user=self.request.user, accepted=True)
+        return Team.objects.filter(
+            Q(captains__exact=self.request.user) | Q(founder=self.request.user) | Q(players__exact=self.request.user))
 
 
 def edit_team_view(request, pk):
     if request.method == 'POST':
         teamobj = get_object_or_404(Team, id=pk)
         form = EditTeamProfileForm(request.POST, request.FILES, instance=teamobj)
+        if request.user in teamobj.captains or request.user is teamobj.founder:
```
Comment (techlover1) [inline teams/views.py:105]: This can be `if not` rather than if, pass, else

File: teams/views.py [hunk 4]
```diff
@@ -83,28 +88,33 @@ class MyTeamsListView(ListView):
...
-            #teamobj.country = form.data['country']
-            #teamobj.image = form.data['image']
-            #teamobj.save()
+            # teamobj.about_us = form.data['about_us']
+            # teamobj.website = form.data['website']
+            # teamobj.twitter = form.data['twitter']
+            # teamobj.twitch = form.data['twitch']
+            # teamobj.country = form.data['country']
+            # teamobj.image = form.data['image']
+            # teamobj.save()
```
Comment (techlover1) [inline teams/views.py:117]: if these commented lines aren't needed they should be deleted

File: teams/views.py [hunk 5]
```diff
@@ -126,22 +136,24 @@ class MyTeamDetailView(DetailView):
...
+        captains = team.captains.all()
         matches_ = Match.objects.filter(awayteam_id=team.id)
         matches__ = Match.objects.filter(hometeam_id=team.id)
         matches = matches_ | matches__
         if not request.user.is_anonymous:
             user = UserProfile.objects.get(user__username=request.user.username)
             if not user.xbl_verified:
-                messages.warning(request, "Xbox Live is not verified")
+                pass
+                # messages.warning(request, "Xbox Live is not verified")
```
Comment (techlover1) [inline teams/views.py:148]: These can just be straight up deleted

File: teams/views.py [hunk 6]
```diff
@@ -167,7 +179,9 @@ def form_valid(self, form):
 
     def get_queryset(self):
         # TO DO switch the filter to the players field not just the founder field.
-        return Team.objects.filter(founder=self.request.user)
+        # TODO: FIX
```
Comment (techlover1) [inline teams/views.py:182]: Resolve this TODO

File: teams/views.py [hunk 7]
```diff
@@ -219,30 +233,39 @@ def post(self, request):
...
         invite = get_invites(form)
-        captains = invite.filter(captain='captain')
-        x = {}
-        for captain in captains:
-            x[captain] = str(captain.user.username)
-        if (request.user == team.founder) or (request.user.username in x.values()):
+        captains = team.captains.all()
+        # x = {}
+        # for captain in captains:
+        #    x[captain] = str(captain.user.username)
```
Comment (techlover1) [inline teams/views.py:239]: Delete unneeded lines

File: teams/views.py [hunk 8]
```diff
@@ -219,30 +233,39 @@ def post(self, request):
...
             try:
                 invitee = UserProfile.objects.get(user__username=form.data['user'])
             except:
                 messages.error(request, "That isn't a valid user")
                 return render(request, 'teams/team_invite_player.html', {'form': form})
             query = invite.filter(user=invitee.user, team=form.data['team'])
             if query.exists():
-                messages.error(request, "That user already has been invited to this team")
-                return redirect('/teams/')
+                messages.error(request, "That user already has already been invited to this team")
```
Comment (techlover1) [inline teams/views.py:248]: "already has already"

File: teams/views.py [hunk 9]
```diff
@@ -309,20 +339,20 @@ def get(self, request, pk):
...
                 return redirect('teams:list')
             else:
-                return redirect('teams:detail', pk)
+                team.players.remove(player)
+                team.save()
+                messages.success(request, 'Removed user %s from team' % player)
+
         else:
-            messages.error(request, "Only the team's founder can remove users")
+            messages.error(request, "Only the team's founder/captain can remove users")
```
Comment (techlover1) [inline teams/views.py:355]: "Only the team's founder or a captain can remove users" would sound better

File: teams/views.py (hunk 4 above)
Comment (mikemaddem) [reply teams/views.py:117]: been there for quite some time. but I'll remove it

File: teams/views.py (hunk 2 above)
Comment (mikemaddem) [reply teams/views.py:37]: done

File: teams/views.py (hunk 6 above)
Comment (mikemaddem) [reply teams/views.py:182]: it has been resolved...?

File: staff/views/teams.py (hunk 1 above)
Comment (mikemaddem) [reply staff/views/teams.py:67]: I don't see it anymore

**Pull request merged**